test/quic_multistream_test.c

```diff
@@ -2843,17 +2843,85 @@ static int script_21_inject_plain(struct helper *h, QUIC_PKT_HDR *hdr,
     switch (h->inject_word1) {
     case OSSL_QUIC_FRAME_TYPE_PATH_CHALLENGE:
     case OSSL_QUIC_FRAME_TYPE_PATH_RESPONSE:
+    case OSSL_QUIC_FRAME_TYPE_RETIRE_CONN_ID:
+        /*
+         * These cases to be formatted properly need a single uint64_t
+         */
         if (!TEST_true(WPACKET_put_bytes_u64(&wpkt, (uint64_t)0)))
             goto err;
         break;
+    case OSSL_QUIC_FRAME_TYPE_MAX_DATA:
+    case OSSL_QUIC_FRAME_TYPE_STREAMS_BLOCKED_UNI:
+    case OSSL_QUIC_FRAME_TYPE_STREAMS_BLOCKED_BIDI:
+    case OSSL_QUIC_FRAME_TYPE_MAX_STREAMS_BIDI:
+    case OSSL_QUIC_FRAME_TYPE_MAX_STREAMS_UNI:
+    case OSSL_QUIC_FRAME_TYPE_DATA_BLOCKED:
+        /*
+         * These cases require a single vlint
+         */
+        if (!TEST_true(WPACKET_quic_write_vlint(&wpkt, (uint64_t)0)))
+            goto err;
+        break;
     case OSSL_QUIC_FRAME_TYPE_STOP_SENDING:
     case OSSL_QUIC_FRAME_TYPE_MAX_STREAM_DATA:
     case OSSL_QUIC_FRAME_TYPE_STREAM_DATA_BLOCKED:
+        /*
+         * These cases require 2 variable integers
+         */
         if (!TEST_true(WPACKET_quic_write_vlint(&wpkt, (uint64_t)0)))
             goto err;
         if (!TEST_true(WPACKET_quic_write_vlint(&wpkt, (uint64_t)0)))
             goto err;
         break;
+    case OSSL_QUIC_FRAME_TYPE_STREAM:
+    case OSSL_QUIC_FRAME_TYPE_RESET_STREAM:
+    case OSSL_QUIC_FRAME_TYPE_CONN_CLOSE_APP:
+        /*
+         * These cases require 3 variable integers
+         */
+        if (!TEST_true(WPACKET_quic_write_vlint(&wpkt, (uint64_t)0)))
+            goto err;
+        if (!TEST_true(WPACKET_quic_write_vlint(&wpkt, (uint64_t)0)))
+            goto err;
+        if (!TEST_true(WPACKET_quic_write_vlint(&wpkt, (uint64_t)0)))
+            goto err;
+        break;
+    case OSSL_QUIC_FRAME_TYPE_NEW_TOKEN:
+        /*
+         * Special case for new token
+         */
+
+        /* New token length, cannot be zero */
+        if (!TEST_true(WPACKET_quic_write_vlint(&wpkt, (uint64_t)8)))
+            goto err;
+
+        /* 8 bytes of token data, to match the above length */
+        if (!TEST_true(WPACKET_put_bytes_u64(&wpkt, (uint64_t)0)))
+            goto err;
```

Comment:
Well, on second thought, the minimum length for a valid token would be one byte,
and the shorter the injection is, the higher the probability that it will fit into the packet.